cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/tableSegment/GetTableSegmentCommand.java

```diff
@@ -0,0 +1,91 @@
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.segmentstore.tableSegment;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.cli.admin.utils.AdminSegmentHelper;
+import io.pravega.client.tables.impl.TableSegmentEntry;
+import io.pravega.client.tables.impl.TableSegmentKey;
+import io.pravega.shared.protocol.netty.PravegaNodeUri;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static io.pravega.cli.admin.utils.SerializerUtils.CHUNK_METADATA_FIELD_MAP;
+import static io.pravega.cli.admin.utils.SerializerUtils.READ_INDEX_BLOCK_METADATA_FIELD_MAP;
+import static io.pravega.cli.admin.utils.SerializerUtils.SEGMENT_METADATA_FIELD_MAP;
+import static io.pravega.cli.admin.utils.SerializerUtils.parseStringData;
+
+public class GetTableSegmentCommand extends TableSegmentCommand {
+
+    /**
+     * Creates a new instance of the GetTableSegmentCommand.
+     *
+     * @param args The arguments for the command.
+     */
+    public GetTableSegmentCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() {
+        ensureArgCount(3);
+
+        final String fullyQualifiedTableSegmentName = getArg(0);
+        final String key = getArg(1);
+        final String segmentStoreHost = getArg(2);
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        AdminSegmentHelper adminSegmentHelper = instantiateAdminSegmentHelper(zkClient);
+        CompletableFuture<List<TableSegmentEntry>> reply = adminSegmentHelper.readTable(fullyQualifiedTableSegmentName,
+                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()),
+                Collections.singletonList(TableSegmentKey.unversioned(getCommandArgs().getState().getKeySerializer().serialize(key).array())),
+                super.authHelper.retrieveMasterToken(), 0L);
+
+        String value = getCommandArgs().getState().getValueSerializer().deserialize(ByteBuffer.wrap(reply.join().get(0).getValue().array()));
+        output("For the given key: %s", key);
+        userFriendlyOutput(value);
+    }
+
+    public static CommandDescriptor descriptor() {
+        return new CommandDescriptor(COMPONENT, "get", "Get the value for the given key in the table.",
+                new ArgDescriptor("qualified-table-segment-name", "Fully qualified name of the table segment to get info from."),
+                new ArgDescriptor("key", "The key to be queried."),
+                new ArgDescriptor("segmentstore-endpoint", "Address of the Segment Store we want to send this request."));
+    }
+
+    private void userFriendlyOutput(String data) {
+        Map<String, String> dataMap = parseStringData(data);
+        if (dataMap.containsKey("key")) {
```

Comment:
Every serializer now implements an abstract method from the base class `AbstractSerializer`, called `getName()` which returns a `String` representing what the serializer deals with. This can be used in the output to identify what kind of data is being queried.